EIPS/eip-7756.md

```diff
@@ -0,0 +1,177 @@
+```
+
+### Summary Line
+
+At the end of execution, the client should print summary info. This summary MUST as a single JSON object.
+
+This info SHOULD have the following members.
+
+| Name        | Type       | Required | Explanation                                            |
+|-------------|------------|----------|--------------------------------------------------------|
+| `stateRoot` | Hex-String | Yes      | Root of the state trie after executing the transaction |
+| `output`    | Hex-String | Yes      | Return values of the function                          |
+| `gasUsed`   | Hex-Number | Yes      | All gas used by the transaction                        |
+| `pass`      | Boolean    | Yes      | If the tx was successful, or if the test passed        |
+| `time`      | Number     | No       | Time in nanoseconds needed to execute the transaction  |
+| `fork`      | String     | No       | Name of the fork rules used for execution              |
+
+*Example*:
+
+```
+{"stateRoot":"0xd4c577737f5d20207d338c360c42d3af78de54812720e3339f7b27293ef195b7","output":"","gasUsed":"0x3","pass":"true","time":141485}
+```
+
+## Rationale
+
+This EIP is an extension of the EIP-3155 tracing features that has been in use for years. Rather than dramatically re-boot the feature, the information was added to the existing traces.
+
+A "mini" trace was contemplated to allow for tracing to be included in tools such as `t8n` and to allow for more efficient RPC tracing calls, but that seemed sufficiently different that it would be a stand-alone EIP rather than an EIP that adds features to the existing tracing capabilities.
+
+## Backwards Compatibility
+
+Clients emitting tracing JSON for uncontained "legacy" contracts will produce a compatible trace, except as outlined below
+
+### Changes from EIP-3155
+
+- The term Client Under Test or CUT has been replaced simply with "client."
+
+### Additions to EIP-3155
+
+- The `immediate` member was added to support the large number of instructions that contain immediate operations.
+  Without this change, users would need bytes of the contracts being executed to rationalize the traces.
+- The `section` and `functionDepth` members were added to support [EIP-4750](./eip-4750) EOF Functions.
+- Added clarification around where `pc` indexes when run in an EOF container.
+
+### Clients
+
+This EIP is fully backward compatible with go-ethereum. OpenEthereum, Besu and Nethermind clients would have to change their JSON output of
+`openethereum-evm` `evmtool` and
+`nethtest` slightly do adhere to the new and stricter specs. New clients would need to implement this change if they want to be part of the differential fuzzing group.
+
+## Test Cases
+
+This is the trace output from the Ethereum Execution Specification Test from one of the parameterized executions of the test `test_eof_functions_contract_call_succeed` in `prague/eip7692_eof_v1/eip3540_eof_v1/test_execution_function.py`.  Memory and return data are disabled.
```

Comment:
You can link to tests directly as long as it matches these regexes:

https://github.com/ethereum/EIPs/blob/46e7845649ee96ecb56f6d76a53b9153f2300b46/config/eipw.toml#L151-L152

Reply:
Done.